meta-resin-common/recipes-containers/docker/docker_git.bb

```diff
@@ -0,0 +1,137 @@
+ . \
+ Also, note that kernel version 3.8 or above is required for proper \
+ operation of the daemon process, and that any lower versions may have \
+ subtle and/or glaring issues. \
+ "
+
+SRCREV = "20f81dde9bd97c86b2d0e33bbbf1388018611929"
+SRCBRANCH = "v1.10.3"
+SRC_URI = "\
+	git://github.com/docker/docker.git;branch=${SRCBRANCH};nobranch=1 \
+	file://docker.service \
+	"
+
+# Apache-2.0 for docker
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=aadc30f9c14d876ded7bedc0afd2d3d7"
+
+S = "${WORKDIR}/git"
+
+DOCKER_VERSION = "1.10.3"
+PV = "${DOCKER_VERSION}+git${SRCREV}"
+
+DEPENDS = " \
+    go-cross \
+    btrfs-tools \
+    "
+
+DEPENDS_append_class-target = "lvm2"
+RDEPENDS_${PN} = "curl git util-linux iptables"
+RRECOMMENDS_${PN} += " kernel-module-dm-thin-pool kernel-module-nf-nat"
+DOCKER_PKG="github.com/docker/docker"
+
+do_configure() {
+}
+
+do_compile() {
+	export PATH=${STAGING_BINDIR_NATIVE}/${HOST_SYS}:$PATH
+
+  export GOHOSTOS="linux"
+  export GOOS="linux"
+
+  case "${TARGET_ARCH}" in
+    x86_64)
+      GOARCH=amd64
+      ;;
+    i586)
+      GOARCH=386
+      ;;
+    arm)
+      GOARCH=${TARGET_ARCH}
+      case "${TUNE_PKGARCH}" in
+        cortexa*)
+          export GOARM=7
+          ;;
+      esac
+      ;;
+    aarch64)
+      # ARM64 is invalid for Go 1.4
+      GOARCH=arm64
+      ;;
+    *)
+      GOARCH="${TARGET_ARCH}"
+      ;;
+  esac
+  export GOARCH
+
+	# Set GOPATH. See 'PACKAGERS.md'. Don't rely on
+	# docker to download its dependencies but rather
+	# use dependencies packaged independently.
+	cd ${S}
+	rm -rf .gopath
+	mkdir -p .gopath/src/"$(dirname "${DOCKER_PKG}")"
+	ln -sf ../../../.. .gopath/src/"${DOCKER_PKG}"
+	export GOPATH="${S}/.gopath:${S}/vendor:${STAGING_DIR_TARGET}/${prefix}/local/go"
+	cd -
+
+	export CGO_ENABLED="1"
+
+	# Pass the needed cflags/ldflags so that cgo
+	# can find the needed headers files and libraries
+	export CGO_CFLAGS="${BUILDSDK_CFLAGS} ${TARGET_CC_ARCH}"
+	export CGO_LDFLAGS="${BUILDSDK_LDFLAGS}  ${TARGET_CC_ARCH} --sysroot=${STAGING_DIR_TARGET}"
+
+	DOCKER_GITCOMMIT="${SRCREV}" \
+	  ./hack/make.sh dynbinary
+}
+
+inherit systemd
+
+SYSTEMD_PACKAGES = "${@base_contains('DISTRO_FEATURES','systemd','${PN}','',d)}"
+SYSTEMD_SERVICE_${PN} = "${@base_contains('DISTRO_FEATURES','systemd','docker.service','',d)}"
+
+do_install() {
+	mkdir -p ${D}/${bindir}
+	cp ${S}/bundles/${DOCKER_VERSION}/dynbinary/docker-${DOCKER_VERSION} \
+	  ${D}/${bindir}/docker
+	cp ${S}/bundles/${DOCKER_VERSION}/dynbinary/dockerinit-${DOCKER_VERSION} \
+	  ${D}/${bindir}/dockerinit
+
+  install -d ${D}${systemd_unitdir}/system
+  install -m 644 ${S}/contrib/init/systemd/docker.* ${D}/${systemd_unitdir}/system
+  # replaces one copied from above with one that uses the local registry for a mirror
+  install -m 644 ${WORKDIR}/docker.service ${D}/${systemd_unitdir}/system
+}
+
+inherit useradd
+USERADD_PACKAGES = "${PN}"
+GROUPADD_PARAM_${PN} = "-r docker"
+
```

Comment:
It creates the /run/docker.sock socket as docker:docker by default, which requires a docker user and group to exist - you can probably override that though